faust/transport/consumer.py

```diff
@@ -1122,6 +1122,9 @@ def _new_offset(self, tp: TP) -> Optional[int]:
             batch = next(consecutive_numbers(acked))
             # remove them from the list to clean up.
             acked[: len(batch)] = []
+            latest_commit_id = self._committed_offset.get(tp, -1)
+            if acked and latest_commit_id and acked[0] <= latest_commit_id:
+                acked[:1] = []
```

Comment:
isn't this same as acked[0] = []

Reply:
Yeah, `acked[:1] = []` is the same as `acked[0] = []`, not sure why that syntax was chosen.